staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,174 @@
+
+	TablePrinter printers.ResourcePrinter
+	OutputFormat *string
+
+	genericclioptions.IOStreams
+	PrintFlags *genericclioptions.PrintFlags
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command
+func NewCmdWhoAmI(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &WhoAmIOptions{
+		OutputFormat: pointer.String(""),
+		IOStreams:    streams,
+		PrintFlags:   genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Check self subject attributes.",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.RunAttributesGet())
+		},
+	}
+
+	allowedFormats := []string{"yaml", "json"}
+	cmd.Flags().StringVarP(o.OutputFormat, "output", "o", *o.OutputFormat, fmt.Sprintf(`Output format. One of: (%s).`, strings.Join(allowedFormats, ", ")))
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *WhoAmIOptions) Complete(f cmdutil.Factory, args []string) error {
+	c, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.AuthClient = c.AuthenticationV1alpha1()
+
+	o.TablePrinter = printers.NewTablePrinter(printers.PrintOptions{AllowMissingKeys: true})
+	return nil
+}
+
+// RunAttributesGet prints all user attributes.
+func (o *WhoAmIOptions) RunAttributesGet() error {
+	sar := &authenticationv1alpha1.SelfSubjectAttributesReview{}
+	response, err := o.AuthClient.SelfSubjectAttributesReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if o.OutputFormat == nil {
+		// This error is prevented on parsing args
+		return fmt.Errorf("unknown output format")
+	}
+
+	switch *o.OutputFormat {
+	case "json":
+		return o.printJSON(response.Status.UserInfo)
+	case "yaml":
+		return o.printYAML(response.Status.UserInfo)
+	default:
+		return o.printTable(response.Status.UserInfo)
+	}
+}
+
+func (o *WhoAmIOptions) printYAML(ui authenticationv1alpha1.UserInfo) error {
+	data, err := yaml.Marshal(ui)
+	if err != nil {
+		return err
+	}
+
+	_, err = fmt.Fprint(o.Out, string(data))
+	return err
+}
+
+func (o *WhoAmIOptions) printJSON(ui authenticationv1alpha1.UserInfo) error {
```

Comment:
@soltysh this doesn't look like the standard marshalling I'm used to seeing.

Reply:
Yeah, this is very wrong.

Reply:
Ok, no objections. I removed this part.